File: api_app/demo/src/main/java/com/example/demo/domain/service/ClientService.java
package com.example.demo.domain.service;

import com.example.demo.domain.entity.Client;
import com.example.demo.domain.repository.ClientRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ClientService {

    final ClientRepository repository;

    public Client save(Client client) {
        return repository.save(client);
    }

    public List<Client> findAll() {
        return repository.findAll();
    }
}

File: api_app/api_connect/lib/domain/service/client_service.dart
import 'package:api_connect/domain/entity/client.dart';
import 'package:api_connect/domain/service/base_service.dart';
import 'package:http/http.dart' as http;

class ClientService extends BaseService {

  ClientService() : super('/client');
  
  Future<List<Client>> findAll() async {
    http.Response response = await http.get(getUri());
    return Client.fromJsonList(response.body);
  }
}